src/Task1/FunctionComponentAndCounter.jsx

```diff
@@ -0,0 +1,30 @@
+import { useState } from "react";
+const FunctionComponentCounter = () => {
+  const [count, setCount] = useState(0);
+
+  return (
+    <div>
+      <div>
+        <button
+          onClick={() => {
+            let temp = count + 1;
+            setCount(temp);
```

Comment:
When you are updating your state in a way that new state is dependent on previous state, you should use callback updates, because state updates in React can be asynchronous and you cannot depend on your current values directly.

You can read more here:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous
https://reactjs.org/docs/hooks-reference.html#functional-updates

P.S.: This comment is related to all of your state updates that depend on the previous state.

Reply:
I found a way to update the state